src/plugins/syscalls/syscalls.cpp

```diff
@@ -128,10 +130,86 @@ static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
     return 0;
 }
 
+static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j;
+
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t*)info->trap->data;
+    syscalls *s = wrapper->sc;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    int nargs = win_syscall_struct[wrapper->syscall_index].num_args;
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    unsigned long size = s->reg_size * nargs;
+    unsigned char buf[size];
+
+    if(s->reg_size==4){ // 32 bit os
+
+        ctx.addr = info->regs->rsp + s->reg_size;  // jump over base pointer
+
+        // multiply num args by 4 for 32 bit systems to get the number of bytes we need
+        // to read from the stack.  assumes standard calling convention (cdecl) for the
+        // visual studio compile.
+        if((size_t)size != vmi_read(vmi, &ctx, buf, size)){
+            return 0;
```

Comment:
If you return 0 here you are going to hang the system (`drakvuf_release_vmi` hasn't been called). Should introduce an `exit` goto label at the end and use it here.